src/libraries/System.Data.Common/src/System/Data/DataTable.cs

```diff
@@ -3516,6 +3516,8 @@ record = NewRecord(-1);
 
         // Prevent inlining so that reflection calls are not moved to caller that may be in a different assembly that may have a different grant set.
         [MethodImpl(MethodImplOptions.NoInlining)]
+        [UnconditionalSuppressMessage("AotAnalysis", "IL3050:RequiresDynamicCode",
```

Comment:
What makes the array type available at runtime?

Reply:
Thanks! Propagating the warnings is a larger task, punting for a later time